src/main/java/com/walmartlabs/x12/rule/TrailerSegmentCountX12Rule.java

```diff
@@ -0,0 +1,108 @@
+package com.walmartlabs.x12.rule;
+
+import com.walmartlabs.x12.SegmentIterator;
+import com.walmartlabs.x12.X12Segment;
+import com.walmartlabs.x12.X12TransactionSet;
+import com.walmartlabs.x12.exceptions.X12ErrorDetail;
+import com.walmartlabs.x12.exceptions.X12ParserException;
+import com.walmartlabs.x12.standard.StandardX12Parser;
+import com.walmartlabs.x12.util.ConversionUtil;
+
+import java.util.List;
+
+public class TrailerSegmentCountX12Rule implements X12Rule {
+
+    /**
+     * check each trailer record and verify 
+     * that the total segments match what 
+     * is reported in the trailer count
+     * 
+     * @param segmentList
+     */
+    @Override
+    public void verify(List<X12Segment> segmentList) {
+        
+        SegmentIterator segments = new SegmentIterator(segmentList);
+
+        int groupHeaders = 0;
+        int groupTrailers = 0;
+        int transactionCount = 0;
+        String currentGroupControlNumber = null;
+        int groupCountOnIeaTrailer = this.findGroupCountOnIeaTrailer(segmentList);
+        
+        while (segments.hasNext()) {
+            X12Segment currentSegment = segments.next();
+            
+            if (StandardX12Parser.GROUP_HEADER_ID.equals(currentSegment.getIdentifier())) {
+                groupHeaders++;
+                currentGroupControlNumber = currentSegment.getElement(6);
+            }
+            
+            if (X12TransactionSet.TRANSACTION_SET_HEADER.equals(currentSegment.getIdentifier())) {
```

Comment:
Should there be TRANSACTION_SET_TRAILER check and validator for that too (validator for ST..SE segment count and control number validator)? 🤔

Reply:
Since the transaction/document is what is being split out, I decided to leave that check to the process that would handle each split out transaction/document individually. 

Similar to the other PR (#87), doing this validation on an individual document/transaction after it has been parsed allows one transaction to be rejected from the EDI message, while another can be processed.